avocado/core/job.py

```diff
@@ -68,17 +68,28 @@ class Job(object):
     along with setup operations and event recording.
     """
 
+    LOG_MAP = {'info': logging.INFO,
+               'debug': logging.DEBUG,
+               'warning': logging.WARNING,
+               'error': logging.ERROR,
+               'critical': logging.CRITICAL}
+
     def __init__(self, args=None):
         """
         Creates an instance of Job class.
 
-        :param args: an instance of :class:`argparse.Namespace`.
+        :param args: the job configuration, usually set by command
+                     line options and argument parsing
+        :type args: :class:`argparse.Namespace`
         """
-        if args is None:
-            args = argparse.Namespace()
-        self.args = args
+        self.args = args or argparse.Namespace()
         self.references = getattr(args, "reference", [])
         self.log = LOG_UI
+        self.loglevel = self.LOG_MAP.get(settings.get_value('job.output',
```

Comment:
What about putting the `get_value` parameters to descriptive variables? It could still be a class attribute as `LOG_MAP`.
```python
class Job(object):
    log_map = {...}
    output_settings_section = 'job.output'
    loglevel_key = 'loglevel'

    def __init__(self, args=None):
        loglevel_set = settings.get_value(output_settings_section, loglevel_key, default='debug')
        self.loglevel = self.LOG_MAP.get(loglevel_set, logging_DEBUG)
        ...
```
It's little bit verbose but it's kinda describe how `settings.get_value` works making explicit what is `job.output` and `loglevel` there.

Reply:
Well, it may just be me, but I find it too verbose indeed.  And when it comes to the configuration settings, if we find ourselves repeating section names and keys, we may want to document those in a module such as `avocado.core.settings` itself IMO.  Something like, in `avocado.core.settings`:

```python

#: This section controls the output of the jobs and will be parsed...
SECTION_JOB_OUTPUT = 'job.output'

#: This is the key for the default log level...
LOG_LEVEL = 'loglevel'
```

But to be honest, I'd like to see something better built on top of both `configparser` and `argparse` so that a clear correlation (by default) of command line options and configuration files (and sections and keys) would be achieved mostly by convention.

Does it make sense to you?

Reply:
Yes. No worries, it was a simple thing. Another idea would be expose an interface like Python dict in settings object. Instead of `settings.get_value('section', 'key')` it would be fine `settings['section']['key']`. IIRC ConfigParser already offers this kind of API, right?